naptime/src/main/scala/org/coursera/naptime/actions/RestAction.scala

```diff
@@ -186,11 +192,15 @@ trait RestAction[RACType, AuthType, BodyType, KeyType, ResourceType, ResponseTyp
           val ctx = new RestContext(body, auth, playRequest, pagination, includes, fields)
 
           def run(): Future[Result] = {
-            val highLevelResponse = safeApply(ctx)
-            highLevelResponse.map { resp =>
+            safeApply(ctx).map { resp =>
+              // log error messages from resources that return a RestError(...)
+              resp match {
+                case RestError(e) if e.httpCode >= 500 =>
+                  logger.error("Server 5xx error response", e)
+                case _ =>
+              }
+
               restEngine.mkResult(rh, fieldsEngine, fields, includes, pagination, resp)
-            } recover {
-              case e: NaptimeActionException => e.result
```

Comment:
I'm guessing this is not a behavior change because wrapping `NaptimeActionException` in `RestError` does something similar?

Reply:
yep, wrapping `NaptimeActionException` allows us to delete this path